src/main/java/sk/uniba/fmph/dcs/game_board/GetCard.java

```diff
@@ -0,0 +1,24 @@
+package sk.uniba.fmph.dcs.game_board;
+
+import sk.uniba.fmph.dcs.stone_age.ActionResult;
+import sk.uniba.fmph.dcs.stone_age.CivilisationCard;
+import sk.uniba.fmph.dcs.stone_age.Effect;
+
+public class GetCard implements EvaluateCivilisationCardImmediateEffect {
+    private boolean used = false;
+    private final CivilisationCard card;
+
+    public GetCard(final CivilisationCard card) {
+        this.card = card;
+    }
+
+    @Override
+    public final ActionResult performEffect(final Player player, final Effect choice) {
+        if (used) {
+            return ActionResult.FAILURE;
+        }
+        used = true;
+        player.playerBoard().giveEndOfGameEffect((card.endOfGameEffect()));
+        return ActionResult.ACTION_DONE;
+    }
+}

```

Comment:
```suggestion
package sk.uniba.fmph.dcs.game_board;

import sk.uniba.fmph.dcs.stone_age.CivilisationCard;
import sk.uniba.fmph.dcs.stone_age.Effect;

import java.util.Collection;
import java.util.Optional;

public class GetCard implements EvaluateCivilisationCardImmediateEffect {
    private final CivilizationCardDeck deck;

    public GetCard(final CivilizationCardDeck deck) {
        this.deck = deck;
    }

    public boolean performEffect(final Player player, final Collection<Effect> choice) {
        Optional<CivilisationCard> cardFromDeck = deck.getTop();
        cardFromDeck.ifPresent(card -> player.playerBoard().giveEndOfGameEffect(card.endOfGameEffect()));
        return true;
    }
}

    }
}

```
prerobene na cardDeck a zatial pretypovanie performEffect na boolean podla planu, ale neviem ci to nieje chyba